test/v1/controllers/AmendBFLossControllerSpec.scala

```diff
@@ -0,0 +1,169 @@
+  val amendBFLoss = AmendBFLoss(lossAmount)
+
+  val amendBFLossResponse = AmendBFLossResponse(Some("XKIS00000000988"), "self-employment", lossAmount, "2019-20")
+
+  val bfLossRequest: AmendBFLossRequest = AmendBFLossRequest(Nino(nino), lossId, amendBFLoss)
+
+  val responseBody: JsValue = Json.parse(
+    s"""
+      |{
+      |    "selfEmploymentId": "XKIS00000000988",
+      |    "typeOfLoss": "self-employment",
+      |    "taxYear": "2019-20",
+      |    "lossAmount": $lossAmount
+      |}
+    """.stripMargin)
+
+  val requestBody: JsValue = Json.parse(
+    s"""
+      |{
+      |  "lossAmount": $lossAmount
+      |}
+    """.stripMargin)
+
+  trait Test {
+    val hc = HeaderCarrier()
+
+    val controller = new AmendBFLossController(
+      authService = mockEnrolmentsAuthService,
+      lookupService = mockMtdIdLookupService,
+      amendBFLossService = mockAmendBFLossService,
+      amendBFLossParser = mockAmendBFLossRequestDataParser,
+      auditService = mockAuditService,
+      cc = cc
+    )
+
+    MockedMtdIdLookupService.lookup(nino).returns(Future.successful(Right("test-mtd-id")))
+    MockedEnrolmentsAuthService.authoriseUser()
+  }
+
+  "amend" should {
+    "return a successful response with header X-CorrelationId and body" when {
+      "the request received is valid" in new Test {
+
+        MockAmendBFLossRequestDataParser.parseRequest(
+          AmendBFLossRawData(nino, lossId, AnyContentAsJson(requestBody)))
+          .returns(Right(bfLossRequest))
+
+        MockAmendBFLossService
+          .amend(AmendBFLossRequest(Nino(nino), lossId, amendBFLoss))
+          .returns(Future.successful(Right(DesResponse(correlationId, amendBFLossResponse))))
+
+        val result: Future[Result] = controller.amend(nino, lossId)(fakePostRequest(requestBody))
+        status(result) shouldBe OK
+        contentAsJson(result) shouldBe responseBody
+      }
+    }
+
+    "return single error response with status 400" when {
+      "the request received failed the validation" in new Test() {
+
+        MockAmendBFLossRequestDataParser.parseRequest(
+          AmendBFLossRawData(nino, lossId, AnyContentAsJson(requestBody)))
+          .returns(Left(ErrorWrapper(None, NinoFormatError, None)))
+
+        val result: Future[Result] = controller.amend(nino, lossId)(fakePostRequest(requestBody))
+        status(result) shouldBe BAD_REQUEST
+        header("X-CorrelationId", result).nonEmpty shouldBe true
+      }
+    }
+    "return a 400 Bad Request with a single error" when {
+
+      val badRequestErrorsFromParser = List(
+        NinoFormatError,
+        RuleIncorrectOrEmptyBodyError,
+        LossIdFormatError,
+        AmountFormatError,
+        RuleInvalidLossAmount
+      )
+
+      val notFoundErrorsFromService = List(
+        NotFoundError
+      )
+
+      badRequestErrorsFromParser.foreach(errorsFromParserTester(_, BAD_REQUEST))
+      notFoundErrorsFromService.foreach(errorsFromServiceTester(_, NOT_FOUND))
```

Comment:
errorsFromServiceTester needs to be checked with FORMAT_NINO and FORMAT_LOSS_ID according to spec